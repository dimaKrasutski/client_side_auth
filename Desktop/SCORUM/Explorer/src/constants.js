import * as THEME_CONSTANTS from 'theme/constants';

export const {
  LOCALE_KEYS,
  LANG_KEYS,
  LOCALE,
  LOCALES,
  DEFAULT_LOCALE,
  DEFAULT_LANG,
} = THEME_CONSTANTS;

export const UNDEFINED_ERROR_ID = 'explorer.error.unknown';

export const MIN_FRACTION_DIGITS = 3;
export const MAX_FRACTION_DIGITS = 9;

export const CURRENCY_USD = 'USD';
export const CURRENCY_EUR = 'EUR';
export const ALL_CURRENCY = [CURRENCY_USD, CURRENCY_EUR];
