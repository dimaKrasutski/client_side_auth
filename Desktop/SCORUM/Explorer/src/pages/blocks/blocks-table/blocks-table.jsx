import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import Table from 'reactstrap/lib/Table';
import { Block } from './block/block';
import { PreloaderBlocks } from '../../../components/preloader/preloader-blocks';

export const BlockTableUI = ({ locale, blocks, inProgress }) => (
  <Table className="mt-3 blocks-table">
    <thead className="thead-light">
      <tr>
        <th>
          <FormattedMessage id="explorer.blocks.table.height" />
        </th>
        <th>
          <FormattedMessage id="explorer.blocks.table.date" />
        </th>
        <th>
          <FormattedMessage id="explorer.blocks.table.hash" />
        </th>
        <th className="text-right">
          <FormattedMessage id="explorer.blocks.table.transactions" />
        </th>
      </tr>
    </thead>
    <tbody>
      {!blocks.length ?
        <PreloaderBlocks /> :
        // TODO: rewrite
        blocks.map(block => <Block key={block[1].block_id} locale={locale} block={block} />)
      }
      {inProgress && <PreloaderBlocks />}
    </tbody>
  </Table>
);

BlockTableUI.propTypes = {
  locale: PropTypes.string.isRequired,
  blocks: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.shape(
          {
            timestamp: PropTypes.string.isRequired,
            block_id: PropTypes.string.isRequired,
            height: PropTypes.number.isRequired,
            transactions: PropTypes.array.isRequired,
          }
        ),
      ])
    )
  ).isRequired,
  inProgress: PropTypes.bool.isRequired,
};

BlockTableUI.defaultProps = {};

export const mapStateToProps = ({ intl, blocks }) => ({
  locale: intl.localeFull,
  blocks: blocks.blocks,
});

export const BlockTable = connect(mapStateToProps)(BlockTableUI);
