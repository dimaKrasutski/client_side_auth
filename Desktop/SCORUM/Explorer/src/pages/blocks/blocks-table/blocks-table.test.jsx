import React from 'react';
import { BlockTableUI } from './blocks-table';
import { LOCALE_KEYS } from '../../../constants';

describe('<BlockTableUI />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<BlockTableUI locale={LOCALE_KEYS.EN_US} blocks={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
