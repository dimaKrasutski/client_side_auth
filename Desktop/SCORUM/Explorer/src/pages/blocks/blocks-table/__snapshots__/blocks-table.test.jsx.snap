// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockTableUI /> should match snapshot 1`] = `
<Table
  className="mt-3 blocks-table"
  responsiveTag="div"
  tag="table"
>
  <thead
    className="thead-light"
  >
    <tr>
      <th>
        <FormattedMessage
          id="explorer.blocks.table.height"
          values={Object {}}
        />
      </th>
      <th>
        <FormattedMessage
          id="explorer.blocks.table.date"
          values={Object {}}
        />
      </th>
      <th>
        <FormattedMessage
          id="explorer.blocks.table.hash"
          values={Object {}}
        />
      </th>
      <th
        className="text-right"
      >
        <FormattedMessage
          id="explorer.blocks.table.transactions"
          values={Object {}}
        />
      </th>
    </tr>
  </thead>
  <tbody>
    <PreloaderBlocks />
  </tbody>
</Table>
`;
